flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -1052,13 +1040,23 @@ protected void recoverAndCommit(FlinkKafkaProducer.KafkaTransactionState transac
                 producer = initTransactionalProducer(transaction.transactionalId, false);
                 producer.resumeTransaction(transaction.producerId, transaction.epoch);
                 producer.commitTransaction();
-            } catch (InvalidTxnStateException | ProducerFencedException ex) {
-                // That means we have committed this transaction before.
+            } catch (InvalidTxnStateException e) {
                 LOG.warn(
-                        "Encountered error {} while recovering transaction {}. "
-                                + "Presumably this transaction has been already committed before",
-                        ex,
-                        transaction);
+                        "Unable to commit recovered transaction ({}) because it's in invalid state. "
+                                + "Most likely the transaction has been aborted for some reason. Please check Kafka logs for more details.",
```

Comment:
```suggestion
                        "Unable to commit recovered transaction ({}) because it's in an invalid state. "
                                + "Most likely the transaction has been aborted for some reason. Please check the Kafka logs for more details.",
```